tests/console/yast2_nfs_server.pm

```diff
@@ -35,12 +35,13 @@ sub run() {
 
     type_string "/sbin/yast2 nfs-server; echo YAST-DONE-\$?- > /dev/$serialdev\n";
 
-    assert_screen([qw/nfs-firewall nfs-config/]);
-    if (match_has_tag('nfs-firewall')) {
-        send_key 'alt-i';
-    }
-
-    assert_screen 'nfs-config';
+    do {
+        assert_screen([qw/nfs-server-not-installed nfs-firewall nfs-config/]);
+        # install missing packages as proposed
+        if (match_has_tag('nfs-server-not-installed') or match_has_tag('nfs-firewall')) {
+            send_key 'alt-i';
+        }
+    } while (match_has_tag('nfs-server-not-installed') or match_has_tag('nfs-firewall'));
```

Comment:
I agree that it looks strange. We want to continue this loop until we hit the right needle, not while we hit the wrong ones :)